schemas/groups/electrical.json

```diff
@@ -238,263 +252,428 @@
         }
-      }
-    },
-
+      }  
+     },
     "inverters": {
       "description": "Data about the Inverter that has both DC and AC quantities",
       "patternProperties": {
         "(^[A-Za-z0-9]+$)": {
           "type": "object",
           "title": "Inverter",
           "description": "DC to AC inverter, one or many, within the vessel",
-          "properties": {
-            "meta": {
-              "$ref": "#/definitions/identity"
-            },
+          "allOf": [{
+            "$ref": "#/definitions/identity"
+            }],   
+          "properties": {   
             "dc": {
               "$ref": "#/definitions/dcQuantities"
             },
             "ac": {
               "$ref": "#/definitions/acQuantities"
             },
-            "mode": {
+            "inverterMode": {
               "type": "object",
               "description": "Mode of inverter",
-              "properties": {
-                "value": {
-                  "type": "string",
-                  "enum": [
-                    "idle",
-                    "inverting",
-                    "disabled",
-                    "standby",
-                    "faulted",
-                    "unknown",
-                    "other"
-                  ]
-                },
-
-                "timestamp": {
-                  "$ref": "../definitions.json#/definitions/timestamp"
+              "allOf": [
+                {
+                  "$ref": "../definitions.json#/definitions/commonValueFields"
                 },
-
-                "$source": {
-                  "$ref": "../definitions.json#/definitions/sourceRef"
+                {
+                "properties": {
+                  "value": {
+                    "type": "string",
+                    "enum": [
+                      "idle",
+                      "inverting",
+                      "disabled",
+                      "standby",
+                      "faulted",
+                      "unknown",
+                      "other"
+                      ]
+                    }
+                  }
                 }
-              }
+              ]
             }
           }
         }
       }
-    },
+    },    
     "chargers": {
-      "description": "Data about the Charger that has both DC and AC quantities",
+      "description": "Data about AC sourced battery charger",
       "patternProperties": {
         "(^[A-Za-z0-9]+$)": {
           "type": "object",
           "title": "Charger",
           "description": "Battery charger",
           "allOf": [{
+            "$ref": "#/definitions/identity"
+          },{
             "$ref": "#/definitions/dcQuantities"
-          }, {
-            "properties": {
-              "meta": {
-                "allOf": [{
-                  "$ref": "#/definitions/identity"
-                }, {
+          },{
+            "$ref": "#/definitions/chargerQuantities"
+          }]
+        }
+      }
+    },
+    "alternator": {
```

Comment:
That was an overall question I had:  Use singular, or plural (but be consistent).  I note a mix throughout Signal-k:  Singular:  Chart, Tank, transmission, Depth, phoneNumber  -- and  plural: Waypoints, Notes, Routes.  (Do not believe there is a plural form of the noun `Solar` w/o making it `Solar Controllers`.  

 Q:  Is there an overall style for Signal-k?